### PR TITLE
Fix anonymous rest argument tracing

### PR DESCRIPTION
**Pull request by st0012**
Methods with `...` parameter doesn't read any argument, so the tracer should just ignore it.

Currently the tracer doesn't handle this case:

```
❯ ruby -Ilib -r debug target.rb
DEBUGGER: Session start (pid: 54807)
[1, 7] in target.rb
=>    1| binding.b(do: "trace pass 1")
      2|
      3| def foo(...)
      4|   block.call
      5| end
      6|
      7| foo(1)
=>#0    <main> at target.rb:1
(rdbg:binding.break) trace pass 1
Enable PassTracer for 1 (enabled)
Traceback (most recent call last):
        5: from target.rb:7:in `<main>'
        4: from target.rb:4:in `foo'
        3: from /Users/st0012/projects/debug/lib/debug/tracer.rb:142:in `block in setup'
        2: from /Users/st0012/projects/debug/lib/debug/tracer.rb:142:in `each'
        1: from /Users/st0012/projects/debug/lib/debug/tracer.rb:153:in `block (2 levels) in setup'
/Users/st0012/projects/debug/lib/debug/tracer.rb:153:in `local_variable_get': wrong local variable name `*' for #<Binding:0x00007ffb7da840f0> (NameError)
```

This makes tracer unusable in Rails apps because it uses `...` in method delegation 

https://github.com/rails/rails/blob/7e8946c4da8b69e3b33a3f16cdfd8923c8756000/activesupport/lib/active_support/core_ext/module/delegation.rb#L198-L199

### CODE REVIEW

File: test/support/utils.rb [hunk 1]
```diff
@@ -38,8 +38,6 @@ def debuggee_backlog test_info
 
     # This method will execute both local and remote mode by default.
     def debug_code(program, boot_options: '-r debug/start', remote: true, &block)
-      check_line_num!(program)
```
Comment (ono-max) [inline test/support/utils.rb:41]: Why did you move this one?

**Review by ono-max — CHANGES_REQUESTED** [review]

File: test/support/utils.rb [hunk 2]
```diff
@@ -71,7 +69,8 @@ def debug_code(program, boot_options: '-r debug/start', remote: true, &block)
       else
         debug_on_local boot_options, TestInfo.new(dup_scenario)
       end
-      assert true
```
Comment (ono-max) [inline test/support/utils.rb:74]: I think this change will confuse users. If they see the following results, they don't know if test works well.

```shell
$ ruby test/debug/quit_test.rb
Tests on local and remote. You can disable remote tests with RUBY_DEBUG_TEST_NO_REMOTE=1.
Loaded suite test/debug/quit_test
Started
...
Finished in 1.047331 seconds.
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
3 tests, 0 assertions, 0 failures, 0 errors, 0 pendings, 0 omissions, 0 notifications
100% passed
--------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------
2.86 tests/s, 0.00 assertions/s
```

Comment (st0012) [reply test/support/utils.rb:74]: Ah I thought this is just a redundant line. But why aren't the previous assertion calls counted?

File: test/support/utils.rb (hunk 1 above)
Comment (st0012) [reply test/support/utils.rb:41]: I think `test/assertion_helper_test.rb` and `test/test_utils_test.rb` are not the main use case for line numbers?
Anyway, I'm not against checking line numbers on them either. But currently I can't adjust my test script easily before they pass because of line numbers. And that's quite frustrating 🙁

File: test/support/utils.rb (hunk 2 above)
Comment (ono-max) [reply test/support/utils.rb:74]: https://github.com/ruby/debug/pull/180#issue-690609876 should help you.

Comment (st0012) [reply test/support/utils.rb:74]: I see it makes sense now. Great work reducing the testing time btw 👍

Comment (st0012) [reply test/support/utils.rb]: I've removed this change.

File: test/support/utils.rb (hunk 1 above)
Comment (st0012) [reply test/support/utils.rb]: I've removed this change as well. But can you find a way to let line checking happen after tests? Otherwise I need to make this change locally every time to I can finish my test script first.

Comment (ono-max) [reply test/support/utils.rb]: Thanks!
`test builder` is fastest way to solve the problem, but you don't wanna use it, right?

**Review by ono-max — APPROVED** [review]

File: test/support/utils.rb (hunk 1 above)
Comment (st0012) [reply test/support/utils.rb]: IMO, test builder should not be a replacement of writing tests in the old fashion way 🙂 
Writing test script in a separate file, run cli commands, and go back to test file to finish the test is just too much switching. 

And even if I do that, I still can't easily modify the test script because it has line numbers. (I know in this case I should update the original file again, but that's confusing).

**Comment by ko1** [comment]
could you rebase it?

**Comment by st0012** [comment]
rebased.

File: lib/debug/tracer.rb [hunk 3]
```diff
@@ -168,11 +169,11 @@ def setup
               end
 
             when :rest
-              next unless name
+              next unless name && name.to_s != "*"
```
Comment (ko1) [inline lib/debug/tracer.rb:172]: FYI: You only need to use `:'*'`

Comment (st0012) [reply lib/debug/tracer.rb:172]: ah, right 🤦 
I've opened https://github.com/ruby/debug/pull/212 for this.
